src/GitHub.App/Services/GitHubContextService.cs

```diff
@@ -0,0 +1,474 @@
+        static readonly Regex windowTitleIssueRegex = new Regex($" · Issue #{issue} · {owner}/{repo}( · GitHub)? - ", RegexOptions.Compiled);
+        static readonly Regex windowTitleBlobRegex = new Regex($"{blobName} at {branch} · {owner}/{repo}( · GitHub)? - ", RegexOptions.Compiled);
+        static readonly Regex windowTitleTreeRegex = new Regex($"{tree} at {branch} · {owner}/{repo}( · GitHub)? - ", RegexOptions.Compiled);
+        static readonly Regex windowTitleBranchesRegex = new Regex($"Branches · {owner}/{repo}( · GitHub)? - ", RegexOptions.Compiled);
+
+        static readonly Regex urlLineRegex = new Regex($"#L(?<line>[0-9]+)(-L(?<lineEnd>[0-9]+))?$", RegexOptions.Compiled);
+        static readonly Regex urlBlobRegex = new Regex($"blob/(?<treeish>[^/]+(/[^/]+)*)/(?<blobName>[^/#]+)", RegexOptions.Compiled);
+
+        static readonly Regex treeishCommitRegex = new Regex($"(?<commit>[a-z0-9]{{40}})(/(?<tree>.+))?", RegexOptions.Compiled);
+        static readonly Regex treeishBranchRegex = new Regex($"(?<branch>master)(/(?<tree>.+))?", RegexOptions.Compiled);
+
+        [ImportingConstructor]
+        public GitHubContextService(IGitHubServiceProvider serviceProvider, IGitService gitService)
+        {
+            this.serviceProvider = serviceProvider;
+            this.gitService = gitService;
+            textManager = new Lazy<IVsTextManager2>(() => serviceProvider.GetService<SVsTextManager, IVsTextManager2>());
+        }
+
+        public GitHubContext FindContextFromClipboard()
+        {
+            var text = Clipboard.GetText(TextDataFormat.Text);
+            return FindContextFromUrl(text);
+        }
+
+        public GitHubContext FindContextFromUrl(string url)
+        {
+            var uri = new UriString(url);
+            if (!uri.IsValidUri)
+            {
+                return null;
+            }
+
+            if (!uri.IsHypertextTransferProtocol)
+            {
+                return null;
+            }
+
+            var context = new GitHubContext
+            {
+                Host = uri.Host,
+                Owner = uri.Owner,
+                RepositoryName = uri.RepositoryName,
+            };
+
+            var repositoryPrefix = uri.ToRepositoryUrl().ToString() + "/";
+            if (!url.StartsWith(repositoryPrefix, StringComparison.OrdinalIgnoreCase))
+            {
+                return context;
+            }
+
+            var subpath = url.Substring(repositoryPrefix.Length);
+
+            (context.Line, context.LineEnd) = FindLine(subpath);
+
+            context.PullRequest = FindPullRequest(url);
+
+            var match = urlBlobRegex.Match(subpath);
+            if (match.Success)
+            {
+                context.TreeishPath = match.Groups["treeish"].Value;
+                context.BlobName = match.Groups["blobName"].Value;
+                return context;
+            }
+
+            return context;
+        }
+
+        public GitHubContext FindContextFromBrowser()
```

Comment:
I thought we'd decided not to put this feature in?

Reply:
This feature is on a back-burner, but the code and the hope is to eventually reactivate it.

The command that uses it `GitHub.OpenFromUrl` has been removed from the UI. It was previously surfaced via `File > Open  > Open from GitHub...`. The command that is surfaced via `GitHub > Open from clipboard is `GitHub.OpenFromClipboard`.

It does make sense to remove it from `GitHub.OpenFromUrl` (which needs to be explicitly bound to a keyboard shortcut).